tests-e2e/argocd/queryscoped_cluster_secrets_test.go

```diff
@@ -0,0 +1,532 @@
+				secret := secretList.Items[index]
+				if strings.HasPrefix(secret.Name, "test-") {
+					err = k8s.Delete(&secret, k8sClient)
+					Expect(err).To(BeNil())
+				}
+			}
+
+			for _, userName := range users {
+				err = application.DeleteArgoCDApplication("test-"+userName, argoCDNamespace)
+				Expect(err).To(BeNil())
+
+				err = fixture.DeleteNamespace(userName, config)
+				Expect(err).To(BeNil())
+			}
+
+			By("cleaning up old ClusterRole/ClusterRoleBinding")
+			for _, userName := range users {
+				clusterRole, clusterRoleBinding := generateReadOnlyClusterRoleandBinding(userName)
+				err = k8sClient.Delete(context.Background(), &clusterRole)
+				if err != nil && !apierr.IsNotFound(err) {
+					Expect(err).To(BeNil())
+				}
+
+				err = k8sClient.Delete(context.Background(), &clusterRoleBinding)
+				if err != nil && !apierr.IsNotFound(err) {
+					Expect(err).To(BeNil())
+				}
+			}
+		}
+
+		// For each user, create: Namespace, ServiceAccounts, Roles/RoleBindings, and optionally, ClusterRole/ClusterRoleBindings.
+		// - The SAs and R/RB are scoped to only access the particular count
+		// - Optionally, a CR/CRB are created to allow Argo CD to read all resources on the cluster (but with no write permissions)
+		createServiceAccountsAndSecretsForTest := func(clusterScopedSecrets bool) {
+			klog := log.FromContext(context.Background())
+
+			By("creating an ServiceAccounts for each user, scoped to only deploy to their namespace, plus cluster-scoped read")
+			userServiceAccountTokens := map[string]string{}
+
+			for _, userName := range users {
+				userToken, err := createNamespaceScopedUserAccount(context.Background(), userName, clusterScopedSecrets, k8sClient, klog)
+				Expect(err).To(BeNil())
+				userServiceAccountTokens[userName] = userToken
+			}
+
+			By("creating Argo CD Cluster secrets for those user's service accounts, scoped to deploy only to their namespace")
+			for userName, userToken := range userServiceAccountTokens {
+
+				jsonString, err := generateClusterSecretJSON(userToken)
+				Expect(err).To(BeNil())
+
+				_, apiURL, err := fixture.ExtractKubeConfigValues()
+				Expect(err).To(BeNil())
+
+				clusterResourcesField := ""
+				namespacesField := ""
+
+				if !clusterScopedSecrets {
+					clusterResourcesField = "false"
+					namespacesField = userName
+				}
+
+				secret := buildArgoCDClusterSecret("test-"+userName, argoCDNamespace, "test-"+userName, apiURL+"?user="+userName,
+					jsonString, clusterResourcesField, namespacesField)
+
+				Expect(secret.Data["server"]).To(ContainSubstring("?"), "the api url must contain a query parameter character, for these tests to be valid")
+
+				err = k8sClient.Create(context.Background(), &secret)
+				Expect(err).To(BeNil())
+
+			}
+		}
+
+		BeforeEach(func() {
+			kubeConfig, err := fixture.GetSystemKubeConfig()
+			Expect(err).To(BeNil())
+
+			k8sClient, err = fixture.GetKubeClient(kubeConfig)
+			Expect(err).To(BeNil())
+
+			cleanupTestResources()
```

Comment:
For this test, since it is kind of an outlier compared to other tests, in that it intentionally creates resources that are not cleaned up by EnsureCleanSlaate, I've added clean to both pre and post. In general, for most tests, I would expect that Pre-only is fine, as you said.